projects/core/src/configurator/commons/facade/configurator-groups.service.ts

```diff
@@ -23,50 +22,17 @@ export class ConfiguratorGroupsService {
     private configuratorGroupStatusService: ConfiguratorGroupStatusService
   ) {}
 
-  /**
-   * Subscribe to update the configuration by the given owner.
-   *
-   * @param owner - Configuration owner
-   */
-  public subscribeToUpdateConfiguration(
-    owner: GenericConfigurator.Owner
-  ): void {
-    // TODO: Cancel previous subscriptions of the configuration state
-    // Set Group Status on each update of the configuration state
-    // This will be called every time something in the configuration is changed, prices,
-    // attributes groups etc.
-    this.configuratorCommonsService
-      .getConfiguration(owner)
-      .subscribe((configuration) =>
-        this.getCurrentGroup(owner)
-          .pipe(take(1))
-          .subscribe((currentGroup) =>
-            this.configuratorGroupStatusService.setGroupStatus(
-              configuration,
-              currentGroup.id,
-              false
-            )
-          )
-      );
-  }
-
   getCurrentGroupId(owner: GenericConfigurator.Owner): Observable<string> {
-    return this.configuratorCommonsService.getUiState(owner).pipe(
-      switchMap((uiState) => {
-        if (uiState && uiState.currentGroup) {
-          return of(uiState.currentGroup);
+    return this.configuratorCommonsService.getConfiguration(owner).pipe(
+      map((configuration) => {
+        if (configuration?.interactionState?.currentGroup) {
+          return configuration.interactionState.currentGroup;
         } else {
-          return this.configuratorCommonsService
-            .getConfiguration(owner)
-            .pipe(
-              map((configuration) =>
-                configuration &&
-                configuration.groups &&
-                configuration.groups.length > 0
-                  ? configuration.groups[0].id
-                  : null
-              )
-            );
+          return configuration &&
```

Comment:
Could you refactor this into an expression using configuration?.groups?.length > 0

Reply:
Done